google-cloud-firestore/src/main/java/com/google/cloud/firestore/Transaction.java

```diff
@@ -61,24 +64,34 @@
     ApiFuture<T> updateCallback(Transaction transaction);
   }
 
+  private final TransactionOptions originalTransactionOptions;
+  @Nullable private final ByteString previousTransactionId;
   private ByteString transactionId;
-  private @Nullable ByteString previousTransactionId;
 
-  Transaction(FirestoreImpl firestore, @Nullable Transaction previousTransaction) {
+  Transaction(
+      FirestoreImpl firestore,
+      TransactionOptions transactionOptions,
+      @Nullable Transaction previousTransaction) {
     super(firestore);
-    previousTransactionId = previousTransaction != null ? previousTransaction.transactionId : null;
+    this.originalTransactionOptions = transactionOptions;
```

Comment:
I am surprised this isn't just `transactionOptions`.

Reply:
Given `TransactionRunner` creates a new instance for each retry, I thought it was useful to name this field to signify it's from the first attempt. Don't feel strongly about it though if you'd prefer I change it.

Reply:
I looks like it is never changed though, at least I cannot find a place that modified this argument in this PR.

Reply:
I'll rename it before merging